rust/sbd-e2e-crypto-client/src/test.rs

```diff
@@ -174,149 +168,33 @@ async fn max_msg_size() {
-        test.conn(Cfg::d().cool(std::time::Duration::from_millis(1))),
-        test.conn(Cfg::d().cool(std::time::Duration::from_secs(5000))),
-    );
-
-    tokio::try_join!(
-        c2.send(c1.pub_key(), b"msg-a"),
-        c3.send(c1.pub_key(), b"msg-b"),
-    )
-    .unwrap();
-
-    let (_, r) = r1.recv().await.unwrap();
-    assert!(r == b"msg-a" || r == b"msg-b");
-    let (_, r) = r1.recv().await.unwrap();
-    assert!(r == b"msg-a" || r == b"msg-b");
-
-    c2.close_peer(c1.pub_key()).await;
-    c3.close_peer(c1.pub_key()).await;
-
-    tokio::time::sleep(std::time::Duration::from_millis(2)).await;
-
-    assert!(c3.send(c1.pub_key(), b"msg-c").await.is_err());
-    assert!(c2.send(c1.pub_key(), b"msg-d").await.is_ok());
+async fn idle_close_connections() {
+    const DUR: std::time::Duration = std::time::Duration::from_millis(500);
 
-    let (_, r) = r1.recv().await.unwrap();
-    assert!(r == b"msg-d");
-}
-
-#[tokio::test(flavor = "multi_thread")]
-async fn cooldown_config_works_from_recv_side() {
-    let test = Test::new().await;
-
-    let ((c1, mut r1), (c2, mut r2), (c3, _r3)) = tokio::join!(
-        test.conn(Cfg::d().cool(std::time::Duration::from_secs(5000))),
-        test.conn(Cfg::d().cool(std::time::Duration::from_millis(1))),
-        test.conn(Cfg::d().cool(std::time::Duration::from_millis(1))),
-    );
-
-    tokio::try_join!(
-        c3.send(c1.pub_key(), b"msg-a"),
-        c3.send(c2.pub_key(), b"msg-b"),
-    )
-    .unwrap();
-
-    let (_, r) = r1.recv().await.unwrap();
-    assert_eq!(b"msg-a", r.as_slice());
-    let (_, r) = r2.recv().await.unwrap();
-    assert_eq!(b"msg-b", r.as_slice());
-
-    c1.close_peer(c3.pub_key()).await;
-    c2.close_peer(c3.pub_key()).await;
-
-    // note we have to close the send side too, or it won't re-init crypto
-    c3.close_peer(c1.pub_key()).await;
-    c3.close_peer(c2.pub_key()).await;
-
-    tokio::time::sleep(std::time::Duration::from_millis(2)).await;
-
-    tokio::try_join!(
-        c3.send(c1.pub_key(), b"msg-c"),
-        c3.send(c2.pub_key(), b"msg-d"),
-    )
-    .unwrap();
-
-    let (_, r) = r2.recv().await.unwrap();
-    assert_eq!(b"msg-d", r.as_slice());
-
-    assert!(
-        tokio::time::timeout(std::time::Duration::from_secs(1), r1.recv())
-            .await
-            .is_err()
-    );
-}
-
-#[tokio::test(flavor = "multi_thread")]
-async fn idle_close_even_if_sending() {
-    let (took, exited_early) =
-        idle_close_even_if_sending_inner(std::time::Duration::from_secs(5000))
-            .await;
-    assert!(!exited_early);
-    let iter_millis = took.as_millis() as u64 / 20;
-    println!("1 iter took: {} millis", iter_millis);
-
-    // let's try to have it close right in the middle of the run
-    let (_took, exited_early) = idle_close_even_if_sending_inner(
-        std::time::Duration::from_millis(iter_millis * 10),
-    )
-    .await;
-    assert!(exited_early);
-}
-
-async fn idle_close_even_if_sending_inner(
```

Comment:
this changes to the more generic `idle_close_connections` as we're handling the recv and send sides separately, paradoxically we now need to track recency of usage from both sides and can now only idle close a connection that has been idle in both directions for the idle timeout.